packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-rpcs.ts

```diff
@@ -0,0 +1,179 @@
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ jsonrpc: '2.0', method: 'eth_blockNumber', params: [], id: 1 }),
+            signal: controller.signal,
+        })
+
+        clearTimeout(timeoutId)
+
+        if (!response.ok) {
+            logger.error(`RPC URL ${rpcUrl} for network ${networkName} responded with status: ${response.status}`)
+            return false
+        }
+
+        const data = await response.json()
+        if (data.result) {
+            return true
+        }
+
+        logger.error(
+            `RPC URL ${rpcUrl} for network ${networkName} responded with invalid data: ${JSON.stringify(data)}`
+        )
+        return false
+    } catch (error) {
+        logger.error(
+            `Validation failed for RPC URL ${rpcUrl} for network ${networkName}: ${error instanceof Error ? error.message : 'An unknown error occurred'}`
+        )
+        return false
+    }
+}
+
+const validateWebSocketRpcUrl = async (rpcUrl: string, timeout: number, networkName: string): Promise<boolean> => {
+    return new Promise((resolve) => {
+        const ws = new WebSocket(rpcUrl)
+        const timeoutId = setTimeout(() => {
+            ws.close()
+            logger.error(`WebSocket connection timed out for RPC URL ${rpcUrl} for network ${networkName}`)
+            resolve(false)
+        }, timeout)
+
+        const cleanup = (success: boolean, message?: string) => {
+            clearTimeout(timeoutId)
+            if (message) {
+                logger.error(message)
+            }
+            ws.close()
+            resolve(success)
+        }
+
+        ws.onopen = () => {
+            const rpcRequest = JSON.stringify({
+                jsonrpc: JSON_RPC,
+                method: 'eth_blockNumber',
+                params: [],
+                id: 1,
+            })
+            ws.send(rpcRequest)
+        }
+
+        ws.onmessage = (event) => {
+            try {
+                const response = JSON.parse(event.data.toString())
+                if (response.jsonrpc === JSON_RPC && !!response.result) {
+                    cleanup(true)
+                } else {
+                    cleanup(false, `Invalid RPC URL ${rpcUrl} for network ${networkName} response: ${event.data}`)
+                }
+            } catch (error) {
+                cleanup(false, `Error parsing RPC response for ${rpcUrl} for network ${networkName}: ${error}`)
+            }
+        }
+
+        ws.onerror = (error) => {
+            cleanup(
+                false,
+                `Validation failed for ${rpcUrl} for network ${networkName}: ${error instanceof Error ? error.message : 'An unknown error occurred'}`
+            )
+        }
+
+        ws.onclose = () => {
+            cleanup(
+                false,
+                `Validation failed for ${rpcUrl} for network ${networkName}: WebSocket connection closed unexpectedly`
+            )
+        }
+    })
+}
```

Comment:
You don't need to rewrite all this logic - it's all already wrapped in a `Provider`.

What you can do is just

```typescript
const providerFactory = createProviderFactory()
const provider = await providerFactory(eid)

// Now get a chain ID or a block number using the provider
```

Reply:
Ah I see what you mean, this is much more simple so I will refactor HTTPS RPCs to be validated using the provider. 

However I noticed this does not work with websocker RPCs, so to validate websocker RPCs, I will use 

```
const wsProvider = new ethers.providers.WebSocketProvider('wss://ethereum-rpc.publicnode.com');
console.log("WSS PROVIDER: ", await wsProvider.getBlockNumber());
```

which works in a very similar way. Will push in next commit!

Reply:
It seemed that `providerFactory` was not recognizing the networks in the provided config, so I used `ethers.JSONRpcProvider` for `https` urls as well